### PR TITLE
AWS auto-unseal acceptance test

### PR DESCRIPTION
**Pull request by tyrannosaurus-becks**

### CODE REVIEW

File: vault/seal/awskms/awskms_test.go [hunk 1]
```diff
@@ -59,6 +59,38 @@ func TestAWSKMSSeal_Lifecycle(t *testing.T) {
 	}
 }
 
+// This test executes real calls. The calls themselves should be free,
```
Comment (calvn) [inline vault/seal/awskms/awskms_test.go]: What's the difference between this and the lifecycle test right above it (`TestAWSKMSSeal_Lifecycle`)?

Comment (calvn) [reply vault/seal/awskms/awskms_test.go]: > This test executes real calls

File: vault/seal/awskms/awskms_test.go [hunk 2]
```diff
@@ -59,6 +59,38 @@ func TestAWSKMSSeal_Lifecycle(t *testing.T) {
...
+// but the KMS key used is generally not free. AWS charges about $1/month
+// per key.
+//
+// To run this test, the following env variables need to be set:
+//   - VAULT_AWSKMS_SEAL_KEY_ID
+//   - AWS_REGION
+//   - AWS_ACCESS_KEY_ID
+//   - AWS_SECRET_ACCESS_KEY
+func TestAccAWSKMSSeal_Lifecycle(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (tyrannosaurus-becks) [reply vault/seal/awskms/awskms_test.go]: How would you feel about this change? https://github.com/hashicorp/vault/pull/5739/commits/4f569b790293874020bef13bd1ee1d04a8964172. It consolidates the repeated code into one block, but stops short of using `t.Run()` and adding test cases. 

I tend to dislike a test-cases approach and test frameworks, which are further along that spectrum. The reason is, I find them hard to read, and hard to run individually during debugging (unless I comment out unwanted test cases, of course). 

If you do feel that test cases are the way to go, I am happy to do it, but wanted to propose this alternative instead.

[the remark this answers is not in the document]
Comment (calvn) [reply vault/seal/awskms/awskms_test.go]: You can actually run sub-test cases individually like so `... -run=TestAccAWSKMSSeal_Lifecycle/mock` if "mock" is the name of the sub-test.

[the remark this answers is not in the document]
Comment (calvn) [reply vault/seal/awskms/awskms_test.go]: We can leave it as-is - I am not too concerned about using `t.Run` or not :)

[the remark this answers is not in the document]
Comment (tyrannosaurus-becks) [reply vault/seal/awskms/awskms_test.go]: OK. Thank you! :-)

**Pull request merged**